gn/build/toolchain/android/android_toolchain.gni

```diff
@@ -14,43 +14,51 @@
 
 import("//build/toolchain/gcc_toolchain.gni")
 
-template("android_arm_toolchain") {
-  gcc_toolchain(target_name) {
-    # TODO : arm64
-    # TODO: Drop API level to 21 after getifaddrs issue fixed
-    _tool_name_root = "armv7a-linux-androideabi24-"
-
-    toolchain_args = {
+template("android_clang_toolchain") {
+  if (current_cpu == "arm") {
```

Comment:
This isn't going to work for multi-arch builds. It only allows creating an Android toolchain with the CPU set in the default toolchain.

It could instead say

```
template("android_clang_toolchain") {
  invoker_toolchain_args = invoker.toolchain_args
  if (invoker_toolchain_args.current_cpu == "arm") {
    # arm logic
  }
}
```
And then we use the template like

```
  android_clang_toolchain("arm") {
    toolchain_args = {
      current_cpu = "arm"
    }
  }

  android_clang_toolchain("arm64") {
    toolchain_args = {
      current_cpu = "arm64"
    }
  }

  android_clang_toolchain("x64") {
    toolchain_args = {
      current_cpu = "x64"
    }
  }

  android_clang_toolchain("x86") {
    toolchain_args = {
      current_cpu = "x86"
    }
  }
```

That'll work for multiarch because we always create toolchains for all 4 architectures.

Reply:
Updated.